### PR TITLE
Feature/#35 ci

### LINKED ISSUE #35: Add Continuous Integration
- [x] Add TravisCI for Linux and OSX builds
- [x] Add AppVeyor for Windows builds

### PR DESCRIPTION
**Pull request by Wulf**
Adds Travis-CI (for linux/osx) and Appveyor (for windows) continuous integration.

Closes #35.

### CODE REVIEW

**Comment by Wulf** [comment]
Going to squash all commits.

**Comment by Wulf** [comment]
@wooorm please look over this.

**Comment by wooorm** [comment]
Other than the below, I’d say try to rebase master onto this, and if the tests pass ok everything’s fine?

>  #warning "Nodehun is no longer being actively maintained. Please email nathanjsweet@gmail.com If you are interested in taking over the project."

Looks like the warning is still triggered?

File: readme.md [hunk 1]
```diff
@@ -1,8 +1,12 @@
 Nodehun
 =======
+[![Build Status](https://travis-ci.org/Wulf/nodehun.svg?branch=master)](https://travis-ci.org/Wulf/nodehun) Unix
+
+[![Build status](https://ci.appveyor.com/api/projects/status/9ky5lws4d191qrui/branch/master?svg=true)](https://ci.appveyor.com/project/Wulf/nodehun/branch/master) Windows
+
```
Comment (wooorm) [inline readme.md]: Drop the `unix` and `windows`, people can infer that from `travis` and `appveyor`

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+version: "{build}"
+
+# Init is run before the repo is cloned
+# Windows line endings.
+init:
```
Comment (wooorm) [inline appveyor.yml]: In my opinion, some of the comments in this file are rather superfluous. I mean, a matrix of `nodejs_versions` with a comment of `Node version to test against` is rather duplicated 😬

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+# Node versions to test against.
+node_js:
+    - "0.10"
+    - "0.12"
+    - "iojs"
+    - "4"
+    - "5"
+    - "6"
+    - "node" # latest
```
Comment (wooorm) [inline .travis.yml]: imo 0.10 and iojs can be dropped (it’s really old / it’s not used often)

File: appveyor.yml (hunk 2 above)
Comment (Wulf) [inline appveyor.yml]: Yeah you're right; I was new to this, so I had comments to remind myself. I'll remove the unnecessary comments.

File: .travis.yml (hunk 3 above)
Comment (Wulf) [inline .travis.yml]: I think we should discuss elsewhere

File: readme.md (hunk 1 above)
Comment (Wulf) [inline readme.md]: Cool cool, the badges look really similar though!

**Comment by wooorm** [comment]
@Wulf What’s going on with AppVeyor?

**Comment by Wulf** [comment]
I added `npm install` in order to obtain tape for the tests but now it fails building the nodehun module itself!

**Comment by wooorm** [comment]
Hmm, any ideas why?

**Comment by Wulf** [comment]
I'm not sure because it builds just fine before the `npm install`. Then afterwards, it breaks... going to play around with it a little.

**Comment by Wulf** [comment]
```
C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\node-gyp\src\win_delay_load_hook.c(34): error C2373: '__pfnDliNotifyHook2': redefinition; different type modifiers [C:\projects\nodehun\build\src\hunspell\hunspell.vcxproj]
  C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\delayimp.h(134): note: see declaration of '__pfnDliNotifyHook2'
```

Okay so this is the culprit.

**Comment by Wulf** [comment]
But honestly it makes no sense because the previous builds (like [this one](https://ci.appveyor.com/project/Wulf/nodehun/build/33/job/8li47mqqumq8ww1v)) build just fine, and then fail building on `npm install`, because `npm install` calls `node-gyp`!

**Comment by Wulf** [comment]
[Here](https://ci.appveyor.com/project/Wulf/nodehun/build/37) - I took out `npm install` and now it can build just fine, but fails because it can't find tape.

**Comment by Wulf** [comment]
Okay I think I got it. There was something wrong with `npm`. Here's a [reference](https://github.com/nodejs/node-gyp/issues/972).

**Comment by Wulf** [comment]
[Two tests](https://ci.appveyor.com/project/Wulf/nodehun/build/47/job/0ont0q9gah823ltg) fail on `iojs`. (windows only)

Took a closer look at AppVeyor, and it's not using all the node versions. I'll have to get to this later.

**Comment by wooorm** [comment]
> iojs

It’s OK if that fails IMO, very few people use it still.

**Comment by wooorm** [comment]
@Wulf Ping! In my opinion we should move forward on this, and ancient Node versions shouldn’t really matter.

**Comment by Wulf** [comment]
Yes I agree. This weekend I should have enough time to get this out of the way. The only problem that's left here is that travis-ci doesn't actually test different node versions right now; they're all using 0.10.46.

**Comment by wooorm** [comment]
Ah, hmm, that’s a biggie.

I actually tried implementing the last commit in nodehun in a project, running on different Node targets in Travis, and I’m getting weird/broken/malloc results: https://travis-ci.org/wooorm/retext-spell/builds/164441264

**Comment by Wulf** [comment]
Oh wow, okay. It didn't fail to build for various node versions when I didn't try downloading/compiling for specific architectures:

https://travis-ci.org/Wulf/nodehun/builds/159014940

Try using this [.travis.yml](https://github.com/Wulf/nodehun/blob/f7f204e643860a039d7f84dc88098b6ba7082ce7/.travis.yml). I didn't have an `npm test` in that commit, so you should add it in.

**Comment by wooorm** [comment]
Weird thing is, it works for me locally (I think I’ve seen a malloc once, not sure anymore)

> Try using this .travis.yml. I didn't have an npm test in that commit, so you should add it in.

Wait, what do you want me to do?

**Comment by Wulf** [comment]
I just checked out your .travis.yml, I think you're missing this:

```
# compiling native node modules
addons:
  apt:
    sources:
      - ubuntu-toolchain-r-test
    packages:
      - g++-4.8
      - g++-4.8-multilib
- gcc-multilib
```

**Comment by Wulf** [comment]
Okay lol you know what else:

From the output of your .travis.yml:
```
gyp: src/hunspell/binding.gyp not found (cwd: /home/travis/build/wooorm/retext-spell/node_modules/nodehun) while loading dependencies of binding.gyp while trying to load binding.gyp
```

It can't find the hunspell binding.gyp. I don't know why that is; I'll look into this on the weekend.

**Comment by wooorm** [comment]
Cool, if you’re tests are running and we can get mine working as well, stuff is good to go! 🚢

**Comment by Wulf** [comment]
Okay these tests should pass, minus that one we talked about. Travis and AppVeyor is all setup.

**Comment by Wulf** [comment]
Travis failed, something to do [this travis issue](https://github.com/travis-ci/travis-ci/issues/5929).

**Pull request merged**